packages/program/src/functions/base-invocation-scope.ts

```diff
@@ -388,4 +392,25 @@ export class BaseInvocationScope<TReturn = any> {
     const transactionRequest = await this.getTransactionRequest();
     return transactionRequest.getTransactionId(chainIdToHash);
   }
+
+  /**
+   * In case the gasLimit and gasPrice are *not* set by the user, this method sets some default values.
+   */
+  async setDefaultTxParams() {
+    const transactionRequest = await this.getTransactionRequest();
+    const { minGasPrice } = this.getProvider().getGasConfig();
+    const { gasUsed } = await this.getTransactionCost();
+
+    const specifiedGasLimit = this.txParameters?.gasLimit || this.hasCallParamsGasLimit;
+    const specifiedGasPrice = this.txParameters?.gasPrice;
+
```

Comment:
Why are we calling `getTransactionCost` again here? This will add at least 1 additional `dryRun` and `estimatePredicates` call to the client. 

This function ( `getTransactionCost` ) is already called by the following `BaseInvocationScope` methods: `call`, `dryRun`, and `simulate`. 

We also already called `getTransactionRequest` ( which will among other things create the script code for the contract call ) within these 3 methods from `BaseInvocationScope`.

What we can do is set `setDefaultTxParams` to be private so it won't be called out of context, and then make it receive `transactionRequest`, `gasLimit`, and `gasPrice` params.

Please let me know if am missing something here.

Reply:
@Torres-ssf good catch about the account methods. Might have to do this for `deployContract` too. Anything else that comes to mind?